cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/dataset/MetadataDataset.java

```diff
@@ -466,22 +464,33 @@ private Metadata getSnapshotBeforeTime(Id.NamespacedId targetId, long timeMillis
   }
 
   /**
-   * Prepares a search query by trimming and also removing whitespaces around the {@link #KEYVALUE_SEPARATOR} and
-   * appending the given namespaceId to perform search in a namespace.
+   * Prepares search terms from the specified search query by
+   * <ol>
+   *   <li>Splitting on {@link #SPACE_SEPARATOR_PATTERN} and trimming</li>
+   *   <li>Handling {@link #KEYVALUE_SEPARATOR}, so searches of the pattern key:value* can be supported</li>
+   *   <li>Prepending the result with the specified namespaceId and {@link Id.Namespace#SYSTEM} so the search can
+   *   be restricted to entities in the specified namespace and {@link Id.Namespace#SYSTEM}.</li>
+   * </ol>t
    *
    * @param namespaceId the namespaceId to search in
    * @param searchQuery the user specified search query
    * @return formatted search query which is namespaced
    */
-  private String prepareSearchQuery(String namespaceId, String searchQuery) {
-    String formattedSearchQuery = searchQuery.toLowerCase();
-    // if this is a key:value search remove  spaces around the separator too
-    if (formattedSearchQuery.contains(KEYVALUE_SEPARATOR)) {
-      // split the search query in two parts on first occurrence of KEYVALUE_SEPARATOR and the trim the key and value
-      String[] split = formattedSearchQuery.split(KEYVALUE_SEPARATOR, 2);
-      formattedSearchQuery = split[0].trim() + KEYVALUE_SEPARATOR + split[1].trim();
+  private Iterable<String> getSearchTerms(String namespaceId, String searchQuery) {
+    List<String> searchTerms = new ArrayList<>();
+    for (String term : Splitter.on(SPACE_SEPARATOR_PATTERN).omitEmptyStrings().trimResults().split(searchQuery)) {
+      String formattedSearchTerm = term.toLowerCase();
+      // if this is a key:value search remove  spaces around the separator too
+      if (formattedSearchTerm.contains(KEYVALUE_SEPARATOR)) {
+        // split the search query in two parts on first occurrence of KEYVALUE_SEPARATOR and the trim the key and value
+        String[] split = formattedSearchTerm.split(KEYVALUE_SEPARATOR, 2);
+        formattedSearchTerm = split[0].trim() + KEYVALUE_SEPARATOR + split[1].trim();
+      }
+      searchTerms.add(namespaceId + KEYVALUE_SEPARATOR + formattedSearchTerm);
+      // also add the system namespace, so entities from system namespace are surfaced in the search results as well
+      searchTerms.add(Id.Namespace.SYSTEM.getId() + KEYVALUE_SEPARATOR + formattedSearchTerm);
```

Comment:
We could add system namespaced search term only if namespace is not equal to system, right?